File: _java/symulacja_organizmow/src/Symulacja/Plansza.java
package Symulacja;

public class Plansza {
    private int N, M; //N to X, M to y
    public Plansza(){
        Set(0,0);
    }
    public Plansza(int N, int M){
        Set(N, M);
    }
    public Plansza(Plansza innaPlansza){
        Set(innaPlansza.GetN(), innaPlansza.GetM());
    }
    public int GetM(){
        return this.M;
    }
    public int GetN(){
        return this.N;
    }
    public int GetLiczbaPol(){
        return N*M;
    }
    public void Set(int N, int M){
        this.N=N;
        this.M=M;
    }
    public void Set(Plansza innaPlansza){
        Set(innaPlansza.GetN(), innaPlansza.GetM());
    }
    @Override
    public String toString(){
        return toString(TrybToString.Ladnie);
    }
    public String toString(TrybToString tryb){
        if(tryb==TrybToString.Ladnie)
            return "[" + N + "x" + M + "]";
        else
            return N + " " + M;
    }
}

File: _c++/symulacja_organizmow/Plansza.cpp
//
//  Plansza.cpp
//  WirtualnySwiat
//
//  Created by Michal Tarnacki on 10/04/2022.
//

#include "Plansza.hpp"

Plansza::Plansza(int n, int m){
    setN(n);
    setM(m);
}
Plansza::Plansza(){
    setN(20);
    setM(20);

}
Plansza::Plansza(Plansza& plansza){
    setN(plansza.GetN());
    setM(plansza.GetM());
}
unsigned int Plansza::GetN() const{
    return N;
}
unsigned int Plansza::GetM() const{
    return M;
}
unsigned int Plansza::GetLiczbaPol() const{
    return M*N;
}
void Plansza::setN(int n){
    if(n<=0) throw "Podano zbyt małe N";
    else N=n;
}
void Plansza::setM(int m){
    if(m<=0) throw "Podano zbyt małe M";
    else M=m;
}
void Plansza::set(int n, int m){
    setM(m);
    setN(n);
}
std::ostream& operator <<(std::ostream&out, Plansza& plansza) {
    out << "Plansza: " << plansza.M << "x" << plansza.N;
    return out;
}
std::ostream& operator <<=(std::ostream&out, Plansza& plansza) {
    out << plansza.M << " " << plansza.N;
    return out;
}
